### DISCUSSION TITLE
Getting the photo filename

### DISCUSSION THREAD

**Discussion by Walimang** [kevinzg/facebook-scraper#832] [Q&A] [answered]
Is there a way I can get the filename of the photo that is being downloaded? Came across a script that downloads all the photos but names them as imagex, imagex+1.. I want the actual name of the photo like 295778817_1094920327897684_8303062189790690596_n.jpg.

`from facebook_scraper import *
import os
import io
import requests
from PIL import Image
import tempfile
def download_image(img_url, out_dir, img_name):
    buffer = tempfile.SpooledTemporaryFile(max_size=1e9)
    r = requests.get(img_url, stream=True)
    if r.status_code == 200:
        downloaded = 0
        filesize = int(r.headers['content-length'])
        for chunk in r.iter_content(chunk_size=1024):
            downloaded += len(chunk)
            buffer.write(chunk)
            print(downloaded/filesize)
        buffer.seek(0)
        i = Image.open(io.BytesIO(buffer.read()))
        i.save(os.path.join(out_dir, img_name), quality=85)
    buffer.close()
count = 0

for img in get_photos('name of the public page without spaces', pages=1000):
    img_name = "".join(['image', str(count), '.jpg'])
    download_image(img['images'][0], "", img_name)`

**Comment by neon-ninja** [comment, marked as answer]
Yes, see https://github.com/kevinzg/facebook-scraper/issues/762

**Reply by Walimang** [reply]
> Yes, see #762

Thank you very much!!
With a little bit of modification, this is working great for me.
